tools/release/python/update_build.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+#
+# Prepare //nlb/BUILD for a Python release.
+
+set -e
+set -o pipefail
+
+# Get the directory of repo root
+script_dir=$(dirname "$(realpath "$0")")
+repo_root=$(realpath "$script_dir/../../..")
+
+function update_deps() {
+    # Query for all Python modules in //nlb/ and add them to //nlb:all_python_modules
+    all_python_modules=$(
+        bazel query '
+            kind(py_binary, //nlb/...) +
+            kind(py_library, //nlb/...) -
+            attr(tags, testdata, //nlb/...)
+        '
+    )
+
+    # Create a temp file to write Buildozer commands
+    temp_file=$(mktemp)
+    for label in $all_python_modules; do
+        echo "add deps $label|//nlb:all_python_modules" >> "$temp_file"
+    done
+
+    # Run Buildozer to update the BUILD file, allow non-zero exit code
+    bazel run --config quiet //tools:buildozer -- -f "$temp_file" || true
+}
+
+function update_completion_data_files() {
+    # Query for all completion files and update the data files in //nlb:nlb_wheel
+    # Completion rules end with `_completions` and are genrules
+    completion_files=$(
+        bazel query 'kind(genrule, //nlb/...) intersect attr(name, _completions, //nlb/...)'
+    )
+
+    # Query the existing data_files keys in //nlb:all_data_files
+    existing_data_files=$(bazel query 'labels(data_files, //nlb:nlb_wheel)')
+
+    # Create a temp file to write Buildozer commands
+    temp_file=$(mktemp)
+    for label in $completion_files; do
+        # Skip if the completion file is already in data_files
+        if echo "$existing_data_files" | grep -q "$label"; then
+            continue
+        fi
+
+        # Extract the name of the completion file from Bazel label (format: //path:label)
+        completion_name=$(echo "$label" | awk -F: '{print $2}' | sed 's/_completions//')
+        # Encode the `:` in the label as `%` for Buildozer
+        # https://github.com/bazelbuild/buildtools/issues/982
+        # (Fixed on main -> needs release + `buildifier_prebuilt` update)
+        escaped_label=$(echo "$label" | sed 's/:/%/')
+        echo "dict_add data_files $escaped_label:data/share/completions/${completion_name}-completion.bash|//nlb:nlb_wheel" >> "$temp_file"
+    done
+
+    # Run Buildozer to update the BUILD file
+    bazel run --config quiet //tools:buildozer -- -f "$temp_file"
+
+    # Replace the `%` back to `:` in the BUILD file
+    sed -i 's/%/:/g' "$repo_root/nlb/BUILD"
+
+    # Run buildifier to format the BUILD file
+    bazel run --config quiet //tools:buildifier -- "$repo_root/nlb/BUILD" || true
```

Comment:
genuinely surprised someone decided to:
a) use `:` as a key-value separator for `BUILD` files
b) "fix" it by using `\\` escapes for `:` in labels instead of picking a better separator
c) document it wrong anyways

?????

Reply:
the documentation on `main` is correct now. I was reading an outdated commit (a & b still stand)